src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -181,53 +180,59 @@ private String getResolvedUri() {
     return resolveUri(uriParameters.getScheme(), uriParameters.getHost().trim(), uriParameters.getPort(), resolvedPath);
   }
 
-  protected String getId() {
-    return getClass().getSimpleName();
-  }
-
-  private Consumer<PollContext.PollItem<String, HttpResponseAttributes>> getPollingItemConsumer(Result<String, HttpResponseAttributes> item) {
+  private Consumer<PollContext.PollItem<String, HttpResponseAttributes>> getPollingItemConsumer(TypedValue<String> fullResponse,
+                                                                                                Result<TypedValue<?>, HttpResponseAttributes> item,
+                                                                                                Serializable watermark) {
     return pollItem -> {
       LOGGER.debug("Setting Result for {}: {}", location.getRootContainerName(), item.getOutput());
-      pollItem.setResult(item);
-      // TODO (HTTPC-178 and HTTPC-179 - idempotency and watermarking)
+      pollItem.setResult(toStringResult(item));
```

Comment:
not following this piece... why `toStringResult`? hasn't the split expression already yielded a String value already?

Reply:
There are a few things here: 
First, the `getItems` returns a stream of results of TypedValue<?> (not string). This is because `split` (from expression manager) returns that type.. and tbh I didn't need to change that (go and change the TypedValue<?> to TypedValue<String>). I use those results also in expressions(e.g. get the id), so I need them as TypedValues. Be it as <?> or <String>. If you think it is better to force it as <String>, no problem. 

But this concrete call is basically to change the result from Result of TypedValue to Result of String, which I need to do regardless of the first paragraph. At some point I tried to leave it as is, but it didn't work well

Reply:
add the explanation as comments in the code